pkg/daemon/ceph/client/upgrade.go

```diff
@@ -216,9 +224,9 @@ func okToStopDaemon(context *clusterd.Context, clusterInfo *ClusterInfo, deploym
 
 // okToContinueOSDDaemon determines whether it's fine to go to the next osd during an upgrade
 // This basically makes sure all the PGs have settled
-func okToContinueOSDDaemon(context *clusterd.Context, clusterInfo *ClusterInfo) error {
+func okToContinueOSDDaemon(context *clusterd.Context, clusterInfo *ClusterInfo, timeout int) error {
 	// Reconciliating PGs should not take too long so let's wait up to 10 minutes
-	err := util.Retry(10, 60*time.Second, func() error {
+	err := util.Retry(1, time.Duration(timeout)*time.Minute, func() error {
```

Comment:
Does this make it so the function no longer checks every minute for updates? Does this mean that OSDs aren't able to finish faster than the default 5 minutes?

Reply:
I have the same concern above, I didn't see this comment soon enough. :)

Reply:
fixed.  Keeping the maxRetires value to 10 and retrying every `timeoutPeriodInSeconds/maxRetries`